gneiss/plot/_blobtree.py

```diff
@@ -0,0 +1,218 @@
+        self.color = color
+
+    def update_items(self):
+        self.item = _DiamondItem(width=self.width, height=self.height,
+                                 label=self.label, color=self.color)
+
+    def _width(self):
+        return self.width
+
+    def _height(self):
+        return self.height
+
+
+def diamondtree(tree, collapsers=None, **kwargs):
+    """ Plots collapsed tree with background coloring and clade coloring.
+
+    This creates collapsed trees similar to the tree plots in the tree of
+    life paper [1]. Rather than plotting all of the leaves, specified
+    subtrees will be replaced by diamonds that are scaled to approximate
+    the depth and the width of the subtree.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        A strictly bifurcating tree defining a hierarchical relationship
+        between all of the features within `table`.
+    collapsers : list of str
+        Names of internal nodes to collapse within the tree.
+    layout : function, optional
+        A layout for formatting the tree visualization. Must take a
+        `ete.tree` as a parameter.
+    labelcolor: str
+        Color of the node labels. (default 'black')
+    bgcolors: dict of str or matplotlib colormap
+        String or function encoding matplotlib colormap for the backgrounds
+        outside of the clades.
+    cladecolors: dict of str or str
+        String or function encoding matplotlib colormap for the colors
+        within the clade faces. (default '#0000FF')
+    depth_scaling : int
+        Scaling factor for height of the subtrees represented by diamonds.
+        (default : 20)
+    breadth_scaling : int
+        Scaling factor for width of the subtrees represented by diamonds.
+        (default : 10)
+    label_size : int
+        Size of nodes labels.
+    mode : str
+        Type of display to show the tree. ('c': circular, 'r': rectangular).
+
+
+    Returns
+    -------
+    ete.Tree
+        ETE tree object that will be plotted.
+    ete.TreeStyle
+        ETE TreeStyle that decorates the tree and heatmap visualization.
+
+    References
+    ----------
+    .. [1] Hug, Laura A., et al. "A new view of the tree of life."
+       Nature Microbiology 1 (2016): 16048.
+    """
+
+    # TODO: Allow for the option to encode labels in different colors
+    # (i.e. pass in a pandas series)
+    params = {'bgcolors': None, 'cladecolors': '#0000FF',
+              'labelcolor': 'black', 'label_size': 10,
+              'depth_scaling': 30, 'breadth_scaling': 6,
+              'mode': 'c',
+              # TODO: Enable layout
+              # layout : function, optional
+              #    A layout for formatting the tree visualization. Must take a
+              #    `ete.tree` as a parameter.
+              # For now just define a null function if no layout is defined
+              # TODO: Learning scaling factors for depth and breadth
+              'layout': lambda x: x}
+    for key in params.keys():
```

Comment:
Cool.  Removing in the next commit.  The reason why I'm unpacking `**kwargs` in the function definition is because I'm anticipating the parameter space to grow up to 50 parameters.  And this isn't too crazy given what we see in [matplotlib plots](http://matplotlib.org/api/pyplot_api.html).  I'm thinking that adding `**kwargs` would make it easier to iteratively add in new parameters. But I'd be curious to hear your thoughts on this.